api/nodes/serializers.py

```diff
@@ -124,6 +127,123 @@ def update(self, instance, validated_data):
+                                 help_text='Highest permission the user has.  Blank input defaults write permission if '
+                                           'user is being added and to current permission if user is being edited.')
+    links = LinksField({
+        'html': 'absolute_url',
+        'nodes': {
+            'relation': Link('users:user-nodes', kwargs={'user_id': '<pk>'})
+        },
+        'detail': Link('nodes:node-contributor-detail',
+                       kwargs={'node_id': '<node_id>', 'user_id': '<pk>'})
+    })
+
+    class Meta:
+        type_ = 'users'
+
+    def absolute_url(self, obj):
+        return obj.absolute_url
+
+    def create(self, validated_data):
+        auth = Auth(self.context['request'].user)
+        node = self.context['view'].get_node()
+        contributor = get_object_or_error(User, validated_data['_id'])
+        # Node object checks for contributor existence but can still change permissions anyway
+        if contributor in node.contributors:
+            raise exceptions.ValidationError('{} is already a validated contributor'.format(contributor.username))
+
+        bibliographic = validated_data['bibliographic']
+        permissions = self.get_permissions_list(validated_data['permission']) \
+            if 'permission' in validated_data else ['read', 'write']
+        node.add_contributor(contributor=contributor, auth=auth, visible=bibliographic, permissions=permissions, save=True)
+        node.reload()
+        contributor.permission = node.get_permissions(contributor)[-1]
+        contributor.bibliographic = node.get_visible(contributor)
+        contributor.node_id = node._id
+        return contributor
+
+    @staticmethod
+    def get_permissions_list(permission):
+        if permission == 'admin':
+            return ['read', 'write', 'admin']
+        elif permission == 'write':
+            return ['read', 'write']
+        elif permission == 'read':
+            return ['read']
+        else:
+            return ['read', 'write']
+
+
+class NodeContributorDetailSerializer(NodeContributorsSerializer):
+    """ Overrides node contributor serializer to make id read only and add addtional methods
+    """
+    id = ser.CharField(read_only=True, source='_id')
+
+    def update(self, instance, validated_data):
+        contributor = instance
+        auth = Auth(self.context['request'].user)
+        node = self.context['view'].get_node()
+        if 'permission' in validated_data and validated_data['permission'] != '':
+            self.set_contributor_permissions(node, validated_data['permission'], contributor, auth)
+        try:
+            node.set_visible(contributor, validated_data['bibliographic'], save=True)
+        except ValueError as e:
+            raise exceptions.ValidationError(e)
+        node.reload()
+        contributor.permission = node.get_permissions(contributor)[-1]
+        contributor.bibliographic = node.get_visible(contributor)
```

Comment:
Not as far as I can tell. The return value from the update is what's getting serialized. Since the user object doesn't have this information as it's complexly defined, we have to add it on. If contributor were its own model, we wouldn't have to do it this way, but a contributor is a User as seen by a Node, so we do. Unless I'm misremembering how we do contributors.